device/delta/x86_64-delta_ag9064-r0/plugins/psuutil.py

```diff
@@ -40,9 +39,10 @@ def get_psu_status(self, index):
             return False
 
         status = 0
+        self.psu_status[1] += str(index) + '_pwr_ok'
         try:
-            p = os.popen(self.psu_status.format(index))
-            content = p.readline().rstrip()
+            p = subprocess.Popen(self.psu_status, stdout=subprocess.PIPE, universal_newlines=True)
```

Comment:
>subprocess.Popen [](http://example.com/codeflow?start=16&length=16)

The intention is to read file to variable. No need to use subprocess at all.

Reply:
Updated.